perllib/FixMyStreet/Cobrand/Southwark.pm

```diff
@@ -30,4 +32,168 @@ sub disambiguate_location {
+    # only Southwark staff may reopen reports
+    my $user = $self->{c}->user_exists ? $self->{c}->user : undef;
+    return 0 if ($user && $user->from_body && $user->from_body->cobrand_name eq $self->council_name);
+    return 1;
+}
+
+
+=item lookup_site_code
+
+Reports sent to Confirm have a "site code" which is usually the USRN of the
+street they're on. For reports made within estates Southwark don't want the
+street USRN to be used, but rather the site code of the estate.
+
+This code ensures the estate site code is used even if the report was made
+on a street that happens to be within an estate.
+
+=cut
+
+sub lookup_site_code {
+    my $self = shift;
+    my $row = shift;
+    my $field = shift;
+
+    if (my $feature = $self->estate_feature_for_point($row->latitude, $row->longitude)) {
+        return $feature->{properties}->{Site_code};
+    }
+
+    return $self->SUPER::lookup_site_code($row, $field);
+}
+
+
+=item lookup_site_code_config
+
+When looking up the USRN of a street where a report was made, A-roads within
+Southwark must be ignored as Southwark's Confirm system is setup to reject
+reports made on such streets. The majority of these street features actually
+have an overlapping non-A-road which will be found and used instead.
+
+=cut
+
+sub lookup_site_code_config {
+    my $host = FixMyStreet->config('STAGING_SITE') ? "tilma.staging.mysociety.org" : "tilma.mysociety.org";
+    return {
+        buffer => 50, # metres
+        url => "https://$host/mapserver/southwark",
+        srsname => "urn:ogc:def:crs:EPSG::27700",
+        typename => "LSG",
+        property => "USRN",
+        accept_feature => sub {
+            # Roads that only have a number, not a name, mustn't be used for
+            # site codes as they're not something Southwark can deal with.
+            # For example "A201", "A3202".
+            my $feature = shift;
+            my $name = $feature->{properties}->{Street_or_numbered_street} || "";
+            return ( $name =~ /^A[\d]+/ ) ? 0 : 1;
+        }
+    };
+}
+
+sub report_new_is_in_estate {
+    my ( $self ) = @_;
+
+    return $self->estate_feature_for_point(
+        $self->{c}->stash->{latitude},
+        $self->{c}->stash->{longitude}
+    ) ? 1 : 0;
+}
+
+sub estate_feature_for_point {
+    my ( $self, $lat, $lon ) = @_;
+
+    my ($x, $y) = Utils::convert_latlon_to_en($lat, $lon, 'G');
+
+    my $host = FixMyStreet->config('STAGING_SITE') ? "tilma.staging.mysociety.org" : "tilma.mysociety.org";
+    my $cfg = {
+        url => "https://$host/mapserver/southwark",
+        srsname => "urn:ogc:def:crs:EPSG::27700",
+        typename => "Estates",
+        filter => "<Filter><Contains><PropertyName>Geometry</PropertyName><gml:Point><gml:coordinates>$x,$y</gml:coordinates></gml:Point></Contains></Filter>",
+    };
+
+    my $features = $self->_fetch_features($cfg, $x, $y);
+    return $features->[0];
+}
+
+# Southwark do not want certain TfL categories to appear, dependent on
```

Comment:
Thanks for the existing docs, would be good to have docs for the other new functionality being added.